modules/openxr/extensions/openxr_opengl_extension.cpp

```diff
@@ -0,0 +1,480 @@
+	opengl_requirements.type = XR_TYPE_GRAPHICS_REQUIREMENTS_OPENGL_KHR;
+	opengl_requirements.next = nullptr;
+
+	XrResult result = xrGetOpenGLGraphicsRequirementsKHR(instance, system_id, &opengl_requirements);
+	if (!openxr_api->xr_result(result, "Failed to get OpenGL graphics requirements!")) {
+		return false;
+	}
+#endif
+
+	if (p_desired_version < opengl_requirements.minApiVersionSupported) {
+		print_line("OpenXR: Requested OpenGL version does not meet the minimum version this runtime supports.");
+		print_line("- desired_version ", OpenXRUtil::make_xr_version_string(p_desired_version));
+		print_line("- minApiVersionSupported ", OpenXRUtil::make_xr_version_string(opengl_requirements.minApiVersionSupported));
+		print_line("- maxApiVersionSupported ", OpenXRUtil::make_xr_version_string(opengl_requirements.maxApiVersionSupported));
+		return false;
+	}
+
+	if (p_desired_version > opengl_requirements.maxApiVersionSupported) {
+		print_line("OpenXR: Requested OpenGL version exceeds the maximum version this runtime has been tested on and is known to support.");
+		print_line("- desired_version ", OpenXRUtil::make_xr_version_string(p_desired_version));
+		print_line("- minApiVersionSupported ", OpenXRUtil::make_xr_version_string(opengl_requirements.minApiVersionSupported));
+		print_line("- maxApiVersionSupported ", OpenXRUtil::make_xr_version_string(opengl_requirements.maxApiVersionSupported));
+	}
+
+	return true;
+}
+
+#ifdef WIN32
+XrGraphicsBindingOpenGLWin32KHR OpenXROpenGLExtension::graphics_binding_gl;
+#elif ANDROID_ENABLED
+XrGraphicsBindingOpenGLESAndroidKHR OpenXROpenGLExtension::graphics_binding_gl;
+#else
+XrGraphicsBindingOpenGLXlibKHR OpenXROpenGLExtension::graphics_binding_gl;
+#endif
+
+void *OpenXROpenGLExtension::set_session_create_and_get_next_pointer(void *p_next_pointer) {
+	XrVersion desired_version = XR_MAKE_VERSION(3, 3, 0);
+
+	if (!check_graphics_api_support(desired_version)) {
+		print_line("OpenXR: Trying to initialize with OpenGL anyway...");
+		//return p_next_pointer;
+	}
+
+	DisplayServer *display_server = DisplayServer::get_singleton();
+
+#ifdef WIN32
+	graphics_binding_gl.type = XR_TYPE_GRAPHICS_BINDING_OPENGL_WIN32_KHR,
+	graphics_binding_gl.next = p_next_pointer;
+
+	graphics_binding_gl.hDC = (HDC)display_server->window_get_native_handle(DisplayServer::WINDOW_VIEW);
+	graphics_binding_gl.hGLRC = (HGLRC)display_server->window_get_native_handle(DisplayServer::OPENGL_CONTEXT);
+#elif ANDROID_ENABLED
+	graphics_binding_gl.type = XR_TYPE_GRAPHICS_BINDING_OPENGL_ES_ANDROID_KHR;
+	graphics_binding_gl.next = p_next_pointer;
+
+	graphics_binding_gl.display = eglGetCurrentDisplay();
+	graphics_binding_gl.config = (EGLConfig)0; // https://github.com/KhronosGroup/OpenXR-SDK-Source/blob/master/src/tests/hello_xr/graphicsplugin_opengles.cpp#L122
+	graphics_binding_gl.context = eglGetCurrentContext();
+#else
+	graphics_binding_gl.type = XR_TYPE_GRAPHICS_BINDING_OPENGL_XLIB_KHR;
```

Comment:
Is this X11 specific? I'm wondering what will happen if a MacOS device attempts to run this code

Reply:
Yes, this is code is X11-specific! AKAIK, OpenXR doesn't work on MacOS. I'll double check the build system and make sure this code isn't compiled on MacOS